app/services/xapi/registered_product_service.rb

```diff
@@ -0,0 +1,36 @@
+module Xapi
+  class RegisteredProductService < Xapi::BaseService
+    def initialize(agent, course)
+      @agent = agent
+      @verb = verb
+      @object = object(course)
+    end
+
+    def set_statement
+      @statement = Xapi.create_statement(actor: @agent, verb: verb, object: @object)
+    end
+
+    private
+
+    def verb
+      Xapi.create_verb(id: 'http://adlnet.gov/expapi/verbs/registered', name: 'registered')
+    end
+
+    def object(course)
+      params = {
+          id: url_helpers.course_url(course, host: course.school.domains.primary.fqdn),
```

Comment:
Is this a good choice of id?
What happens if 2 different users are registered for the same course at the same school?
Maybe create an id by hashing that route with `@agent` email and a timestamp?

Reply:
Nvm, that would be more of object type or tag, no?